src/Ocelot/DependencyInjection/OcelotBuilder.cs

```diff
@@ -144,6 +144,9 @@ public OcelotBuilder(IServiceCollection services, IConfiguration configurationRo
             Services.TryAddSingleton<ISecurityOptionsCreator, SecurityOptionsCreator>();
             Services.TryAddSingleton<ISecurityPolicy, IPSecurityPolicy>();
 
+            // Features
+            Services.AddHeaderRouting();
```

Comment:
@ggnaegi So, this adds the feature. Additionally, you've restructured the code and introduced a new **MessageInvokerPool** feature at line 120: `Services.AddOcelotMessageInvokerPool();`.
I'm interested in expanding on this concept. It would be beneficial to have a feature-based design for Ocelot in the future.